CONTRIBUTING.md

```diff
@@ -123,6 +120,24 @@ To add a test to the Flutter package, create a file whose name
 ends with `_test.dart` in the `packages/flutter/test` directory. The
 test should have a `main` function and use the `test` package.
 
+Working with flutter tools
+--------------------------
+
+The flutter shell tools itself is built when you run `flutter` for the first time and each time
+you run `flutter upgrade`. If you want to alter and re-test the shell tool's behavior itself,
+delete `bin/cache/flutter_tools.*` so that the shell tool will be rebuilt from Dart sources
+in `packages/flutter_tools` the next time you run `flutter`.
+
+flutter_tools' tests run inside the dart command line VM rather than recursively in the
+flutter shell. To run the test:
+
+* `cd packages/flutter_tools`
+* `dart test/all.dart`
+
+The pre-built shell tool runs with the observatory (thus debugging) off by default.
```

Comment:
Understood. Terminology collision :) I was thinking shell as in bash shell but it's taken. Removed shell in the non sky_shell sense here and below. 

@Hixie, so adding --observe flag specifically turns the observatory on while still in release mode. Is that the right interpretation? Or does --observe turn it back to debug mode? 

Tentatively reworded. PTAL.